tests/unit/dom.ts

```diff
@@ -237,5 +237,169 @@ registerSuite({
+							dom.containsClass(null, 'test');
+						});
+					},
+
+					'invalid class'() {
+						assert.throws(function () {
+							dom.containsClass(targetElement, 'te est');
+						});
+					},
+
+					'null class'() {
+						assert.doesNotThrow(function () {
+							dom.containsClass(targetElement, null);
+						});
+					}
+				},
+
+				removeClass: {
+					afterEach() {
+						targetElement.className = '';
+					},
+
+					'remove single class'() {
+						dom.addClass(targetElement, 'test');
+						assert.include(targetElement.className, 'test');
+						dom.removeClass(targetElement, 'test');
+						assert.notInclude(targetElement.className, 'test');
+					},
+
+					'remove multiple classes'() {
+						dom.addClass(targetElement, 'test1', 'test2');
+						assert.include(targetElement.className, 'test1');
+						assert.include(targetElement.className, 'test2');
+						dom.removeClass(targetElement, 'test1', 'test2');
+						assert.notInclude(targetElement.className, 'test1');
+						assert.notInclude(targetElement.className, 'test2');
+					},
+
+					'null node'() {
+						assert.doesNotThrow(function () {
+							dom.removeClass(null, 'test');
+						});
+					},
+
+					'invalid class'() {
+						assert.throws(function () {
+							dom.removeClass(targetElement, 'te est');
+						});
+					},
+
+					'null class'() {
+						assert.doesNotThrow(function () {
+							dom.removeClass(targetElement);
+						});
+					},
+
+					'remove nonexistent class'() {
+						assert.doesNotThrow(function () {
+							dom.removeClass(targetElement, 'random');
+						});
+					},
+
+					'remove mutiple nonexistent classes'() {
+						assert.doesNotThrow(function () {
+							dom.removeClass(targetElement, 'random', 'random1');
+						});
+					}
+				},
+
+				toggleClass: {
+					'toggling existing class removes it'() {
+						dom.addClass(targetElement, 'test');
+						assert.include(targetElement.className, 'test');
+						dom.toggleClass(targetElement, 'test');
+						assert.notInclude(targetElement.className, 'test');
+					},
+
+					'toggling nonexistent class adds it'() {
+						dom.toggleClass(targetElement, 'test');
+						assert.include(targetElement.className, 'test');
+					},
+
+					'toggling class using force = true'() {
+						dom.toggleClass(targetElement, 'test', true);
+						assert.include(targetElement.className, 'test');
+						dom.toggleClass(targetElement, 'test', true);
+						assert.include(targetElement.className, 'test');
+					},
+
+					'toggling class using force = false'() {
+						dom.toggleClass(targetElement, 'test', false);
+						assert.notInclude(targetElement.className, 'test');
+						dom.addClass(targetElement, 'test');
+						dom.toggleClass(targetElement, 'test', false);
+						assert.notInclude(targetElement.className, 'test');
+					}
+				}
+			};
+		}
+
+		return {
+			HTMLElement: cssSuite(document.createElement('div')),
+			SVGElement: cssSuite(document.createElement('svg'))
```

Comment:
What methods are you thinking of?  I particularly poked around with SVG yesterday while talking to Paul about these APIs, and other than the fact that IE doesn't support native `classList` at all on them, they should work (they expose `className`).

Incidentally, if we didn't have to support IE9 *or* SVG, we could've saved some time here because we could always rely on native `classList`, we'd just have to call the native methods multiple times in browsers that don't support multi-argument add/remove, and would have to do some finagling for `toggle` in browsers that don't support the force argument.